CMakeLists.txt

```diff
@@ -86,6 +86,27 @@ CONFIGURE_FILE(
 	"${nanoflann_BINARY_DIR}/nanoflann.pc" @ONLY IMMEDIATE )
 
 
+
+# Set relative install directories
+set(INSTALL_INCLUDE_DIR "include")
+set(INSTALL_PKGCONFIG_DIR "lib${LIB_SUFFIX}/pkgconfig")
+set(INSTALL_CMAKE_DIR "lib${LIB_SUFFIX}/cmake/nanoflann")
+set(INSTALL_COPYRIGHT_DIR "share/doc/libnanoflann-dev")
+
+
+# Generate the cmake config and cmake config-version file:
+SET(RELATIVE_INCLUDE_DIR "../../../include")
+SET(CONFIG_INCLUDE_DIRS "\${nanoflann_CMAKE_DIR}/${RELATIVE_INCLUDE_DIR}")
+
```

Comment:
I have adapted this from what is recommended by CMake itself (see https://cmake.org/Wiki/CMake/Tutorials/How_to_create_a_ProjectConfig.cmake_file).  Although more modern versions exist, this one should also work with old CMake 2.6 and is used in several other projects.

The variable `${PROJECT_NAME}_CMAKE_DIR` is defined inside `${PROJECT_NAME}Config.cmake` and needed to make the installed package relocatable (see "Closing Remarks" in the CMake Tutorial). One should definitely go with this version.